core/sql/optimizer/SynthType.cpp

```diff
@@ -5207,6 +5207,18 @@ const NAType *Translate::synthesizeType()
          err4106arg = SQLCHARSETSTRING_UTF8;
        break;
 
+     case GBK_TO_UTF8:
+       if (translateSource->getCharSet() == CharInfo::GBK || translateSource->getCharSet() == CharInfo::UnknownCharSet )
+         charsetTarget = CharInfo::UTF8;
```

Comment:
I need to remove the 'unknowncharset' checking. It is wrong.
In the beginning, I assumed Trafodion will treat the column charset of hive table as unknown, since it seems to me Hive doesn't have charset info in the column definition. So I add this checking, but it turns out Trafodion will treat Hive column charset as 'HIVE_DEFAULT_CHARSET', so it should never be 'unknown'. I will remove this checking, it is confusing and wrong assumption.